### PR TITLE
Add cloud-credentials config

### PR DESCRIPTION
**Pull request by stencell**
##### SUMMARY

This config is very simple. It will use a set of master credentials for a sandbox or other type of account and create a set of credentials specific for a user that will be removed upon retirement. It does not create any cloud infra - networks, instances, storage, etc. - since all that is required is the credentials.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
cloud-credentials

### CODE REVIEW

File: ansible/configs/cloud-credentials/README.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+* ec2
+
+== AgnosticV Requirements
+
+In your agnosticV config, you should have the following set to enable provisioning of the sandboxy account
+
+----
+agnosticv_meta:
+  sandboxed: true
+----
```
Comment (fridim) [inline ansible/configs/cloud-credentials/README.adoc:20]: Let's add the equivalent for Babylon:

```yaml
__meta__:
  aws_sandboxed: true
```


Also, make sure we set the variable to force aws-nuke.

```yaml
agnosticv_meta:
  sandboxed: true
  nuke_sandbox: true
```

In babylon, all sandboxes are marked for cleanup (with aws-nuke)  see https://github.com/redhat-gpte-devopsautomation/babylon_aws_sandbox/blob/master/tasks/release.yaml#L47.

File: ansible/configs/cloud-credentials/post_software.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      msg: "Step 005 Post Software"
+
+    - name: Add AWS sandbox credentials if enabled
+      when:
+      - cloud_provider == 'ec2'
+      block:
+      - name: Set facts for AWS credentials
+        set_fact:
+          student_access_key_id: "{{ cloudformation_out_final.stack_outputs.StudentUserAccessKey }}"
+          student_secret_access_key: "{{ cloudformation_out_final.stack_outputs.StudentUserSecretAccessKey }}"
```
Comment (fridim) [inline ansible/configs/cloud-credentials/post_software.yml:20]: I would find it more logical to have this portion of code in the `post_infra.yml` playbook

**Comment by stencell** [comment]
@fridim Changes made as suggested.

File: ansible/configs/cloud-credentials/post_infra.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        agnosticd_user_info:
+          msg: "{{ item }}"
+          data:
+            aws_access_key_id: "{{ student_access_key_id }}"
+            aws_secret_access_key: "{{ student_secret_access_key }}"
+            aws_subdomain_base_suffix: "{{ subdomain_base_suffix }}"
+          loop:
+            - "aws_access_key_id: {{ student_access_key_id }}"
+            - "aws_secret_access_key: {{ student_secret_access_key }}"
+            - "aws_sandbox_zone: {{ subdomain_base_suffix }}"
```
Comment (jkupferer) [inline ansible/configs/cloud-credentials/post_infra.yml:33]: Setting `msg` and `data` should be separate tasks. Setting `data` should not have a loop.

**Review by jkupferer — CHANGES_REQUESTED** [review]

**Comment by fridim** [comment]
@stencell CI failed with:

```
FAIL ........ Galaxy roles cloud-credentials/sample_vars_ec2.yml
ERROR! conflicting action statements: debug, msg
The error appears to be in '/home/travis/build/redhat-cop/agnosticd/ansible/configs/cloud-credentials/pre_infra.yml': line 10, column 7, but may
be elsewhere in the file depending on the exact syntax problem.
The offending line appears to be:
  tasks:
    - debug:
      ^ here
ERROR: InvocationError for command /home/travis/build/redhat-cop/agnosticd/tests/static/syntax-check.sh (exited with code 2)
```

Indentation is wrong

**Comment by stencell** [comment]
@fridim @jkupferer The issues you both called out have been corrected. If you can verify, release, and merge I would appreciate it!

**Review by jkupferer — APPROVED** [review]

**Pull request merged**